extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -737,26 +735,41 @@ public void onFetchPage(@Nonnull final Downloader downloader)
             youtubePlayerResponse = playerResponse;
         }
 
-        JsonObject playabilityStatus = (playerResponse == null ? youtubePlayerResponse : playerResponse)
-                .getObject("playabilityStatus");
+        final JsonObject playabilityStatus = (playerResponse == null ? youtubePlayerResponse
+                : playerResponse).getObject("playabilityStatus");
+
+        checkPlayabilityStatus(youtubePlayerResponse, playabilityStatus);
+
+        nextResponse = getJsonPostResponse("next", body, localization);
+
+        streamingData = playerResponse.getObject("streamingData");
+        if (hasOtfStreams() || isCipherProtectedContent()) {
+            fetchAndroidMobileJsonPlayer(contentCountry, localization, videoId, stsKnown);
+        }
+    }
+
+    private void checkPlayabilityStatus(final JsonObject youtubePlayerResponse,
+                                        JsonObject playabilityStatus) throws ParsingException {
         String status = playabilityStatus.getString("status");
         // If status exist, and is not "OK", throw the specific exception based on error message
         // or a ContentNotAvailableException with the reason text if it's an unknown reason.
         if (status != null && !status.equalsIgnoreCase("ok")) {
             playabilityStatus = youtubePlayerResponse.getObject("playabilityStatus");
             status = playabilityStatus.getString("status");
-
             final String reason = playabilityStatus.getString("reason");
 
             if (status.equalsIgnoreCase("login_required")) {
                 if (reason == null) {
                     final String message = playabilityStatus.getArray("messages").getString(0);
-                    if (message != null && message.equals("This is a private video. Please sign in to verify that you may see it.")) {
+                    if (message != null && message.equals(
+                            "This is a private video. Please sign in to verify that you may see it.")) {
```

Comment:
I'd change this for future-proof-ness
```suggestion
                    if (message != null && message.contains("private")) {
```